CHANGELOG.md

```diff
@@ -0,0 +1,38 @@
+# Changelog
```

Comment:
Existential question time!  We've generally stopped using `CHANGELOG.md` files in our node.js repos in favor of release notes on GitHub.  The rationale was that we didn't want to maintain the data in two places.  

@ofrobots - what are your feelings about this as a pattern?

Reply:
For Node we standardized the process as per go/package-publishing (internal link) where the release notes are kept in the GitHub release pages. See also go/npm-at-google (internal link). Greenkeeper and Renovate (and users who are tracking the repo as well) are notified of the release notes as well. Greenkeeper and Renovate specifically can include the release notes in the PRs they end up opening in dependent projects which makes it a lot easier for dependent modules to understand what changed rather than trying to find where the list of changes is in our repos.

Reply:
Here's some pros:

In many other languages this is great to have as a file because it provides history and context *outside* of GitHub. For example, in Python we often have downstream re-packagers (Debian) who greatly appreciate our inclusion of a CHANGELOG.

Releasetool also currently plucks the GitHub release description from this file once this PR is merged, so it gives y'all an opportunity to review the release description via normal code review means. (we can ofc change this)

Reply:
@ofrobots this wouldn't be *instead of* release notes, but *in addition to*. You can see how this looks in Python [here](https://github.com/GoogleCloudPlatform/google-cloud-python/releases/tag/monitoring-0.30.0).

The main benefit I see over the existing release process for node is as I mentioned above:

> Releasetool also currently plucks the GitHub release description from this file once this PR is merged, so it gives y'all an opportunity to review the release description via normal code review means. (we can ofc change this)

What do you think?

Reply:
To help me visualize the workflow... when does the release tag get published to github? who creates the release notes on github? I am guessing that the CI is still responsible publishing the npm module?

Reply:
fwiw, I *can* make releasetool stage a release just like y'all do today, it's just those are much more difficult IMO to *review*.

Reply:
I am pro having the release notes in two places, as long as we have a tool (like releasetool...) that's responsible for doing the work.

![why not both](https://i.giphy.com/media/3o85xIO33l7RlmLR4I/giphy.webp)

Reply:
> I am pro having the release notes in two places, as long as we have a tool (like releasetool...) that's responsible for doing the work.

That's my POV as well. Releasetool handles this extremely well out-of-the-box.